### PR TITLE
feat(tx): mark genesis blocks and transactions

### PR DESCRIPTION
**Pull request by giovanecosta**
https://github.com/HathorNetwork/hathor-explorer/issues/46

Genesis block/transactions pages now will look like this:
image
image

### CODE REVIEW

File: src/components/TxData.js [hunk 1]
```diff
@@ -431,14 +431,21 @@ class TxData extends React.Component {
       );
     }
 
+    const renderGenesisMark = (hash) => {
+      console.log(hash)
```
Comment (pedroferreira1) [inline src/components/TxData.js:435]: Debug log

File: src/constants.js [hunk 2]
```diff
@@ -25,12 +25,18 @@ export const DASHBOARD_BLOCKS_COUNT = 6;
...
 
 export const GENESIS_BLOCK = [
-  '000000a0f82cfee5431e03b071364970861ffa1b0633f73ca7f462987ec34195'
+  '000000a0f82cfee5431e03b071364970861ffa1b0633f73ca7f462987ec34195', // Default
+  '0000033139d08176d1051fb3a272c3610457f0c7f686afbe0afe3d37f966db85', // Testnet
+  '000006cb93385b8b87a545a1cbb6197e6caff600c12cc12fc54250d39c8088fc' // Mainnet
 ]
 
 export const GENESIS_TX = [
   '000000831cff82fa730cbdf8640fae6c130aab1681336e2f8574e314a5533849',
```
Comment (pedroferreira1) [inline src/constants.js:34]: I think we can remove the first two transactions. It's an old genesis test

File: src/constants.js [hunk 3]
```diff
@@ -25,12 +25,18 @@ export const DASHBOARD_BLOCKS_COUNT = 6;
 export const DASHBOARD_TX_COUNT = 6;
 
 export const GENESIS_BLOCK = [
-  '000000a0f82cfee5431e03b071364970861ffa1b0633f73ca7f462987ec34195'
```
Comment (pedroferreira1) [inline src/constants.js:28]: I think we can remove the first hash. It's an old genesis test

**Review by pedroferreira1 — APPROVED** [review]

File: src/constants.js [hunk 4]
```diff
@@ -25,12 +25,18 @@ export const DASHBOARD_BLOCKS_COUNT = 6;
 export const DASHBOARD_TX_COUNT = 6;
 
 export const GENESIS_BLOCK = [
```
Comment (msbrogli) [inline src/constants.js:27]: I don't think we should mix hashes from mainnet and testnet. Maybe we should have two config files: one for the mainnet and another for testnet. @giovanecosta @pedroferreira1 What do you think?

**Review by msbrogli — CHANGES_REQUESTED** [review]

File: src/constants.js (hunk 4 above)
Comment (pedroferreira1) [reply src/constants.js:27]: Maybe we can start with two different variables in the same constants file? The lib works like this nowadays and we still need to implement this issue (https://github.com/HathorNetwork/hathor-wallet-lib/issues/161) which discusses exactly having a configuration file and being able to easily customize it.

To be honest, the best approach would be for the explorer to get this information (genesis hashes) from the lib in the future but I feel we can start with this solution for now.

Comment (msbrogli) [reply src/constants.js:27]: Ok. We can split into two different variables (one for the testnet and another one for the mainnet).

**Review by msbrogli — CHANGES_REQUESTED** [review]
You need to sign your commits and fix the commit messages.

File: src/constants.js (hunk 4 above)
Comment (giovanecosta) [reply src/constants.js:27]: I was looking into the code trying to understand if we could use environment variables to do it, so, we can keep one constant and the explorer agnostic from which network it is connecting. But i realized that we deploy it on Cloudfront, so, no env vars :sweat_smile:

In this case, i agree with two different variables and i'll make a function into `utils.helpers` that checks the network and return if the tx is genesis or not

**Review by pedroferreira1 — APPROVED** [review]

File: src/constants.js (hunk 4 above)
Comment (msbrogli) [reply src/constants.js]: The site is static so there should be a way to inject code based on envvars in compile time.

File: src/utils/helpers.js [hunk 5]
```diff
@@ -156,6 +157,41 @@ const helpers = {
...
+   * Checks if transaction is a Genesis transaction
+   *
+   * @param {string} hash Hash of transaction
+   *
+   * @return {boolean} true if is genesis, false otherwise
+   * @memberof Helpers
+   * @inner
+   */
+  isGenesisTx(hash) {
+    if (hathorLib.network.getNetwork().name === 'mainnet') {
```
Comment (msbrogli) [inline src/utils/helpers.js:171]: Move it to a separate method `isMainnet()`.

**Review by msbrogli — CHANGES_REQUESTED** [review]

**Review by msbrogli — APPROVED** [review]

**Pull request merged**